### DISCUSSION TITLE
Where is the good starting point to read prisma codebase to understand entire prisma architecture?

### DISCUSSION THREAD

**Discussion by thilllon** [prisma/prisma#16096] [Q&A] [answered]
Hi,
Prisma has greatly improved my productivity, which I am very grateful for. I want to read the Prisma code to get a better understanding, but I'm lost on where to start. Is it good to read `packages/engines` module first? It might be a silly question. Any tips would be great, I would appreciate it if you could let me know!

(I've already read the documentation a lot, and of course it's been very helpful. I really appreciate the hard work of the authors.)

**Comment by nurul3101** [comment, marked as answer]
Hey @thilllon 👋 
You might want to go through this [video](https://www.youtube.com/watch?v=a51KX3vxB4E) to understand Prisma architecture in a holistic way. The video demonstrates the major parts of Prisma architecture and was helpful to me in getting an overview of the codebase.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
